uni-stark/src/verifier.rs

```diff
@@ -55,24 +55,20 @@ where
         .collect_vec();
 
     // Check that the random commitments are/are not present depending on the ZK setting.
-    if SC::Pcs::ZK {
-        // If ZK is enabled, the prover should have random commitments.
-        if opened_values.random.is_none() || commitments.random.is_none() {
-            return Err(VerificationError::RandomizationError);
-        }
-        // If ZK is not enabled, the prover should not have random commitments.
-    } else if opened_values.random.is_some() || commitments.random.is_some() {
+    // - If ZK is enabled, the prover should have random commitments.
+    // - If ZK is not enabled, the prover should not have random commitments.
+    if (opened_values.random.is_some() && commitments.random.is_some()) != SC::Pcs::ZK {
```

Comment:
I'm not convinced that the logic is unchanged here. Assume that `SC::Pcs::ZK` was false, `opened_values.random.is_some()` was `True` and `commitments.random.is_some()` was `False.

Then the initial code would error but the new code would not. (In particular the error happens when `opened_values.random.is_some()` and `commitments.random.is_some()` are not equal).

I think the right fix might be

```suggestion
    if (opened_values.random.is_some() != SC::Pcs::ZK) || (commitments.random.is_some()) != SC::Pcs::ZK) {
```
Maybe double check this though.